### PR TITLE
USPS endicia configuration added in endicia carrier form

### PR DESCRIPTION
**Pull request by usudaysingh**

### CODE REVIEW

File: carrier.py [hunk 1]
```diff
@@ -21,6 +54,24 @@ def __setup__(cls):
         if selection not in cls.carrier_cost_method.selection:
             cls.carrier_cost_method.selection.append(selection)
 
+        cls._error_messages.update({
```
Comment (prakashpp) [inline carrier.py]: remove

File: carrier.py [hunk 2]
```diff
@@ -21,6 +54,24 @@ def __setup__(cls):
...
+            'endicia_credentials_required':
+                'Endicia settings on endicia configuration are incomplete.',
+        })
+
+    def get_endicia_credentials(self):
```
Comment (prakashpp) [inline carrier.py]: you dont need this method anymore

File: stock.py [hunk 3]
```diff
@@ -551,18 +530,11 @@ def default_buy_postage(self, data):
...
-        EndiciaConfiguration = Pool().get('endicia.configuration')
-
+        Carrier = Pool().get('carrier')
         default = {}
-        endicia_credentials = EndiciaConfiguration(1).get_endicia_credentials()
-
         buy_postage_api = BuyingPostageAPI(
             request_id=Transaction().user,
             recredit_amount=self.start.amount,
-            requesterid=endicia_credentials.requester_id,
```
Comment (prakashpp) [inline stock.py]: requesterid=self.carrier.requester_id,

File: carrier.py [hunk 4]
```diff
@@ -1,19 +1,52 @@
...
-from trytond.model import ModelSQL, ModelView, fields
+from decimal import Decimal
+from trytond import backend
+from trytond.model import ModelSQL, ModelView, fields, ModelSingleton
 from trytond.pool import PoolMeta, Pool
 from trytond.transaction import Transaction
+from trytond.pyson import Eval
+
+REQUIRED_IF_ENDICIA = {
+    'required': Eval('carrier_cost_method') == 'endicia',
```
Comment (priyankarani) [inline carrier.py]: also add for invisible when not endicia

File: carrier.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+            <field name="model">carrier</field>
+            <field name="type">form</field>
+            <field name="inherit" ref="carrier.carrier_view_form" />
+            <field name="name">carrier_form</field>
+        </record>
+        
+        <record model="ir.action.act_window" id="act_endicia_configuration_form">
+            <field name="name">Endicia Carrier</field>
+            <field name="res_model">carrier</field>
+        </record>
```
Comment (priyankarani) [inline carrier.xml]: not needed, act window is already there in parent module

File: carrier.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+        <menuitem parent="stock.menu_configuration" id="endicia_config"
+        action="act_endicia_configuration_form" sequence="5" icon="tryton-list"/>
```
Comment (priyankarani) [inline carrier.xml]: not needed

**Comment by priyankarani** [comment]
you haven't fixed tests yet. Please fix them too.

File: carrier.py [hunk 7]
```diff
@@ -1,19 +1,53 @@
...
-class Carrier:
+class Carrier(ModelSingleton):
     "Carrier"
     __name__ = 'carrier'
 
+    # endicia start from here
+    account_id = fields.Char('Account Id', states=REQUIRED_IF_ENDICIA)
+    requester_id = fields.Char('Requester Id', states=REQUIRED_IF_ENDICIA)
+    passphrase = fields.Char('Passphrase', states=REQUIRED_IF_ENDICIA)
+    is_test = fields.Boolean('Is Test')
```
Comment (priyankarani) [inline carrier.py]: add states here too

File: carrier.py [hunk 8]
```diff
@@ -1,19 +1,53 @@
...
 __metaclass__ = PoolMeta
 
 
-class Carrier:
+class Carrier(ModelSingleton):
     "Carrier"
     __name__ = 'carrier'
 
+    # endicia start from here
+    account_id = fields.Char('Account Id', states=REQUIRED_IF_ENDICIA)
```
Comment (priyankarani) [inline carrier.py]: add depends for every field

depends = ['carrier_cost_method']

**Pull request closed without merging**